src/gui/creds/httpcredentialsgui.cpp

```diff
@@ -113,7 +117,7 @@ void HttpCredentialsGui::showDialog()
             + QLatin1String("<br>");
     }
 
-    QInputDialog *dialog = new QInputDialog();
+    QInputDialog *dialog = new QInputDialog(qobject_cast<Application *>(qApp)->gui()->settingsDialog());
```

Comment:
Do we need the cast here because qApp returns a QCoreApplication?
Maybe use QApplication::instance() instead?

Reply:
Its QApplication vs Application....